### ISSUE TITLE
PyTorch model  operator within a DALI pipeline

### ISSUE THREAD

**Issue by giobus75** [NVIDIA/DALI#4832]
### Describe the question.

Hi, 
I have a PyTorch model that performs an image transformation (like an image domain transfer). I was wondering if it is possible to use that model as an operator within a DALI pipeline along with other preprocessing operations (for example after a resize and a crop).
Thank you
Giovanni

### Check for duplicates

- [X] I have searched the [open bugs/issues](https://github.com/NVIDIA/DALI/issues) and have found no duplicates for this bug report

**Comment by JanuszL** [comment]
Hi @giobus75,

Thank you for reaching out. In that case, I think you can try using [the Python operator](https://docs.nvidia.com/deeplearning/dali/user-guide/docs/examples/custom_operations/python_operator.html) and invoke the PyTorch model there. Although DALI was not designed to execute models inside.
If it is about the inference you can consider using [DALI TRITON backend](https://github.com/triton-inference-server/dali_backend).

**Comment by giobus75** [comment]
Thank you for your quick answer. 
The use case I'm working on is a neural network training where images need to be preprocessed within the DALI pipeline before being used as input for a resnet50. One of the preprocessing operations, as I said before, is a domain transfer model running on the same GPU of the resnet50.
I was thinking about using the Python operator or ```nvidia.dali.plugin.pytorch.fn.the torch_python_function``` but I don't know how much it can affect the performance. Should I expect an important impact from that point of view?

**Comment by JanuszL** [comment]
> I was thinking about using the Python operator or nvidia.dali.plugin.pytorch.fn.the torch_python_function but I don't know how much it can affect the performance

The Python function doesn't run asynchronously and it won't nicely overlap with the rest of the processing. However, the network itself could be big enough and not run with the rest of the training in parallel so the overall performance loss due to the way DALI handles the execution of the python code could be negligible.

**Comment by giobus75** [comment]
Thank you @JanuszL 
Bye 
Giovanni

**Comment by JanuszL** [comment]
Hi @giobus75,

Now the Python function can execute asynchronously to the pipeline in the pipeline mode. You can try it out and see how that works for you.

**Comment by giobus75** [comment]
Hi @JanuszL , thank you, I'll try it as soon as possible!
